common/app/common/LinkTo.scala

```diff
@@ -20,19 +20,17 @@ trait LinkTo extends Logging {
 
   lazy val host = Configuration.site.host
 
-  private val ProdURL = "^http://www.theguardian.com/.*$".r
-  private val GuardianUrl = "^(http://www.theguardian.com)?(/.*)?$".r
+  private val ProdURL = "^http[s]?://www.theguardian.com/.*$".r
+  private val GuardianUrl = "^(http[s]?://www.theguardian.com)?(/.*)?$".r
```

Comment:
Group with () screws up the match such that the expected value that is checked later needs additional work to cope with its absence. I could drop the square brackets completely but I think isolating the 's' makes it clearer what we are expecting.